### PR TITLE
[SPARK-15028][SQL] Remove HiveSessionState.setDefaultOverrideConfs

### PR DESCRIPTION
**Pull request by rxin**
## What changes were proposed in this pull request?

This patch removes some code that are no longer relevant -- mainly HiveSessionState.setDefaultOverrideConfs.
## How was this patch tested?

N/A

### CODE REVIEW

**Comment by rxin** [comment]
cc @yhuai

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionState.scala [hunk 1]
```diff
@@ -44,8 +44,6 @@ private[hive] class HiveSessionState(sparkSession: SparkSession)
    */
   lazy val metadataHive: HiveClient = sharedState.metadataHive.newSession()
 
-  setDefaultOverrideConfs()
```
Comment (yhuai) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionState.scala]: Also remove this method?

Comment (rxin) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionState.scala]: removed

**Comment by yhuai** [comment]
lgtm

**Comment by yhuai** [comment]
Merging to master.

**Pull request closed without merging**